src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -156,16 +156,20 @@ public void CheckConfigEnvParsingThrowExceptions(string invalidEnvVarName)
             Assert.ThrowsException<DataApiBuilderException>(() => JsonSerializer.Deserialize<StubJsonType>(json, options));
         }
 
+        [DataRow("\"notsupporteddb\"", "")]
+        [DataRow("\"mssql\"", "\"notsupportedconnectionstring\"")]
         [TestMethod("Validates that JSON deserialization failures are gracefully caught.")]
-        public void TestDeserializationFailures()
+        public void TestDataSourceDeserializationFailures(string dbType, string connectionString)
         {
             string configJson = @"
 {
     ""data-source"": {
-        ""database-type"": ""notsupporteddb""
-     }
+        ""database-type"": " + dbType + @",
+        ""connection-string"": " + connectionString + @"
+    },
+    ""entities"":{ }
 }";
-            Assert.IsFalse(RuntimeConfigLoader.TryParseConfig(configJson, out RuntimeConfig deserializedConfig));
+            Assert.IsFalse(RuntimeConfigLoader.TryParseConfig(configJson, out RuntimeConfig deserializedConfig, replaceEnvVar: true));
```

Comment:
what is replaceEnvVar doing for the test? could yo add a test description/displaynames for the data rows?

Reply:
Sure I added a comment. This is just used to get into the portion of the code that does the post-processing, which is what we are testing for the connection string.

Reply:
if we don't replace environment variables, the connectionstringBuilder check / exception won't be hit? wouldn't an unreplace env_var be an invalid connection string?

Reply:
the connection string builder is called within this call that happens, and this call is only hit for MSSQL with replace environment variables being true
![image](https://github.com/Azure/data-api-builder/assets/93220300/3846dfe6-f037-4604-964d-1365b3f901da)

Reply:
gotcha, add that to the PR description too and/or note that in the test description comment for the test you changed from mssql->pgsql